pkg/reconciler/workload/synctargetexports/synctarget_indexes.go

```diff
@@ -0,0 +1,82 @@
+/*
+Copyright 2022 The KCP Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package synctargetexports
+
+import (
+	"fmt"
+
+	"github.com/kcp-dev/logicalcluster/v2"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	"k8s.io/client-go/tools/clusters"
+
+	apisv1alpha1 "github.com/kcp-dev/kcp/pkg/apis/apis/v1alpha1"
+	workloadv1alpha1 "github.com/kcp-dev/kcp/pkg/apis/workload/v1alpha1"
+	reconcilerapiexport "github.com/kcp-dev/kcp/pkg/reconciler/workload/apiexport"
+)
+
+// indexAPIExportsByAPIResourceSchemasFunc is an index function that maps an APIExport to its spec.latestResourceSchemas.
+func indexAPIExportsByAPIResourceSchemas(obj interface{}) ([]string, error) {
+	apiExport, ok := obj.(*apisv1alpha1.APIExport)
+	if !ok {
+		return []string{}, fmt.Errorf("obj is supposed to be an APIExport, but is %T", obj)
+	}
+
+	ret := make([]string, len(apiExport.Spec.LatestResourceSchemas))
+	for i := range apiExport.Spec.LatestResourceSchemas {
+		ret[i] = clusters.ToClusterAwareKey(logicalcluster.From(apiExport), apiExport.Spec.LatestResourceSchemas[i])
+	}
+
+	return ret, nil
```

Comment:
does this work? Which kind of comparison does an indexer do? By value (which would be a pointer here) or deep equal?

Reply:
nvmd. You value is string.